src/Swarm/Game/Step.hs

```diff
@@ -862,17 +862,26 @@ execConst c vs k = do
         flagRedraw
         return $ Out (VString (newRobot' ^. robotName)) k
       _ -> badConst
+    -- run can take both types of text inputs
+    -- with and without file extension as in
+    -- "./path/to/file.sw" and "./path/to/file"
     Run -> case vs of
       [VString fileName] -> do
-        mf <- liftIO $ readFileMay (into fileName)
+        mf <- liftIO $ readFileMay (into filePath)
 
-        f <- mf `isJustOr` cmdExn Run ["File not found:", fileName]
+        f <- mf `isJustOr` cmdExn Run ["File not found:", filePath]
 
         t <-
           processTerm (into @Text f) `isRightOr` \err ->
-            cmdExn Run ["Error in", fileName, "\n", err]
+            cmdExn Run ["Error in", filePath, "\n", err]
 
         return $ initMachine' t empty k
+       where
+        hasExtension = T.isSuffixOf ".sw" fileName
```

Comment:
This could surprise a user that does not know about the `.sw` extension and has for example a file `moves` or `zigzag.swarm` - maybe consider adding a `doesFileExist` above `readFileMay`? :slightly_smiling_face:

Reply:
Perhaps we should pick the first existing file from [fileName, fileName <> ".sw"], and keep the provided fileName in the error message.

Reply:
I've made the required changes.